pallets/asset-manager/src/migrations.rs

```diff
@@ -365,8 +365,19 @@ where
 						used_weight = used_weight.saturating_add(2 * db_weights.write);
 					}
 
+					// This is checked in case UnitsWithAssetType runs first
+					if let Some(units) = AssetTypeUnitsPerSecond::<T>::get(&asset_type) {
```

Comment:
```suggestion
					if let Some(units) = AssetTypeUnitsPerSecond::<T>::take(&asset_type) {
```
could take and get rid of the `remove` below but it is equivalent logic